raiden/utils/__init__.py

```diff
@@ -36,6 +36,9 @@ def sha3(data):
         TypeError: This function does not accept unicode objects, they must be
         encoded prior to usage.
     """
+    # FIXME: check that only bytes objects get here
+    if isinstance(data, str):
+        data = data.encode()
```

Comment:
I'm against this change, we need to keep a coherent core that uses binary, this is making it easy to mess up data types.